Package const folded graph's cubin file

yushangdi opened the pull request:
Summary: We need to pacakge const folded graph's cubin file into the final .pt2 package.

Fix https://github.com/pytorch/pytorch/issues/152067

Test Plan:
```
buck run fbcode//mode/dev-nosan //caffe2/test/inductor:test_aot_inductor -- -r test_constant_folding_cuda
```

Differential Revision: D73626480




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

henrylhtsang commented:
Very noob question: What does the const graph look like? And what kind of info does it have

yushangdi commented:
> Very noob question: What does the const graph look like? And what kind of info does it have

@henrylhtsang The const folded graph just contains the `relu` and the `add`. In the case of cuda, it uses a triton kernel to compute them.

`relu` and `add` are in the const folded graph because they do not rely on the input. 

I don't think it has any special information, just a normal GraphModule.

yushangdi committed:
Package const folded graph's cubin file (#152145)

Summary:

We need to pacakge const folded graph's cubin file into the final .pt2 package.

Test Plan:
```
buck run fbcode//mode/dev-nosan //caffe2/test/inductor:test_aot_inductor -- -r test_constant_folding_cuda

buck run fbcode//mode/dev-nosan //caffe2/test/inductor:test_aot_inductor -- -r test_duplicate_constant_folding

buck run fbcode//mode/dev-nosan //caffe2/test/inductor:test_aot_inductor -- -r test_output_misaligned
```

Reviewed By: henrylhtsang

Differential Revision: D73626480
